x-pack/legacy/plugins/monitoring/index.ts

```diff
@@ -0,0 +1,135 @@
+  home: ['plugins/monitoring/register_feature'],
+  styleSheetPaths: resolve(__dirname, 'public/index.scss'),
+};
+
+const validConfigOptions: string[] = [
+  'xpack.monitoring.ui.enabled',
+  'xpack.monitoring.kibana.collection.enabled',
+  'xpack.monitoring.max_bucket_size',
+  'xpack.monitoring.min_interval_seconds',
+  'kibana.index',
+  'xpack.monitoring.show_license_expiration',
+  'xpack.monitoring.ui.container.elasticsearch.enabled',
+  'xpack.monitoring.ui.container.logstash.enabled',
+  'xpack.monitoring.tests.cloud_detector.enabled',
+  'xpack.monitoring.kibana.collection.interval',
+  'xpack.monitoring.elasticsearch.hosts',
+  'xpack.monitoring.elasticsearch',
+  'xpack.monitoring.xpack_api_polling_frequency_millis',
+  'server.uuid',
+  'server.name',
+  'server.host',
+  'server.port',
+  'xpack.monitoring.cluster_alerts.email_notifications.enabled',
+  'xpack.monitoring.cluster_alerts.email_notifications.email_address',
+  'xpack.monitoring.ccs.enabled',
+  'xpack.monitoring.elasticsearch.logFetchCount'
+];
+
+/**
+ * Invokes plugin modules to instantiate the Monitoring plugin for Kibana
+ * @param kibana {Object} Kibana plugin instance
+ * @return {Object} Monitoring UI Kibana plugin object
+ */
+export const monitoring = (kibana: LegacyPluginApi): ArrayOrItem<LegacyPluginSpec> => {
+
+  return new kibana.Plugin({
+    require: ['kibana', 'elasticsearch', 'xpack_main'],
+    id: 'monitoring',
+    configPrefix: 'xpack.monitoring',
+    publicDir: resolve(__dirname, 'public'),
+    config: configDefaults,
+    uiExports,
+    deprecations,
+
+    init(server: Server) {
+      const serverConfig = server.config();
+      const { injectUiAppVars, getOSInfo, plugins } = server as (typeof server & { getOSInfo?: any });
+      const log = (...args: Parameters<typeof server.log>) => server.log(...args);
+      const route = (...args: Parameters<typeof server.route>) => server.route(...args);
+      const expose = (...args: Parameters<typeof server.expose>) => server.expose(...args);
+      const serverFacade = {
+        config: () => ({
+          get: (key: string) => {
+            if (validConfigOptions.includes(key)) {
+              return serverConfig.get(key);
+            }
+            throw `Unknown key '${key}'`;
+          }
+        }),
+        injectUiAppVars,
+        log,
+        getOSInfo,
+        events: {
+          on: ((...args: Parameters<typeof server.events.on>) => server.events.on(...args))
+        },
+        route,
+        expose,
+        _hapi: server,
+        // @ts-ignore
+        _kbnServer: this.kbnServer as KibanaServer
+      };
+
+      const serverPlugins = plugins as Partial<typeof plugins> & { infra?: InfraPlugin };
+      const { xpack_main, elasticsearch, infra } = serverPlugins;
+      const { usageCollection } = server.newPlatform.setup.plugins;
+      new Plugin().setup(serverFacade, { xpack_main, elasticsearch, infra, usageCollection, });
```

Comment:
for the next steps: migration guide recommends to separate the api available via core, plugins and depreacted legacy API ` const demoSetup = new Plugin().setup(coreSetup, pluginsSetup, __LEGACY);`
https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#switch-to-new-platform-services